lib/randomfileaccess.js

```diff
@@ -134,10 +135,26 @@ class RandomFileAccess {
      */
     async write(path, range, buffer, totalSize) {
         const { fileHandles } = this[PRIVATE];
-        const handle = await getOrOpen(fileHandles, path, "w", totalSize);
+        const handle = await getOrOpen(fileHandles, path, FileFlags.WRITEONLY, totalSize);
         return handle.write(buffer, 0, buffer.length, range.low);
     }
 
+    /**
+     * Write a section to a file. WARNING: this method will read
+     * the entire contents of the stream into memory; use with
+     * extreme caution.
+     * 
+     * @param {String|URL} path Path of the file to write to
+     * @param {SubRange} range Range of bytes to write
+     * @param {Stream} stream Stream to write.
+     * @param {Number} totalSize Total size of the file written
+     * @returns {WriteResponse} Bytes written and the buffer 
+     */
+    async writeStream(path, range, stream, totalSize) {
+        const buffer = await streamToBuffer(stream);
+        return this.write(path, range, buffer, totalSize);
+    }
```

Comment:
As mentioned above, the intent is to have `Transfer` deal with block I/O so `RandomFileAccess` should just deal with blocks. I'd move `streamToBuffer` to the `Transfer` javascript file and then call `RandomFileAccess.write` from `Transfer`.